fastparse/jvm/src/test/scala/fastparse/Main.scala

```diff
@@ -0,0 +1,105 @@
+package fastparse
+
+import javax.sound.midi.MidiSystem
+
+/**
+  * Exercise MidiParse by using it to parse and then play some music!
+  *
+  * - sbt "~fastparseJVM/test:run canon.mid"
+  * - sbt "~fastparseJVM/test:run go.mid"
+  * - sbt "~fastparseJVM/test:run ctend.mid"
+  */
+object Main {
+  def main(args: Array[String]): Unit = {
+    println("Parsing Midi...")
+    val bytes = MidiTests.readResourceBytes("/" + args(0))
+    MidiTests.hexBytes(bytes)
+    val parsed = MidiParse.midiParser.parse(bytes).get.value
+    println("format:\t"+parsed.format)
+    println("tickDiv:\t"+parsed.tickDiv)
+    play(parsed)
+  }
+  def play(midi: Midi) = {
+
+
+    import Midi._, MidiData._
+    // 120bpm, in microseconds-per-quarter-beat
+    var currentTempo = 1000000 * 120 / 60 / 4
+    println("defaultTempo:\t"+currentTempo)
+//    midi.tracks(0).collect{ case Midi.MetaEvent.Tempo(n) => n}
+
+    val synth = MidiSystem.getSynthesizer()
+    val channels = synth.getChannels()
+    synth.open()
+    case class TrackInfo(var savedTicks: Int, var track: List[(Int, Midi.TrackEvent)]){
+      def tillNext = track.head._1 - savedTicks
+    }
+    val allTracks = midi.tracks.map(_.toList).map(TrackInfo(0, _))
+
+    println("playing...")
+    while({
```

Comment:
Seems like code-duplication, but I think it can cause more bulky code if you try to rewrite it.